extensions/spring-xd-extension-http/src/main/java/org/springframework/integration/x/http/NettyHttpInboundChannelAdapter.java

```diff
@@ -139,10 +151,35 @@ public NettyHttpInboundChannelAdapter(int port) {
 	}
 
 	public NettyHttpInboundChannelAdapter(int port, boolean ssl) {
+		/* begin whu */
+		this(port, "/", ssl);
+		/* end whu */
+	}
+
+	/* begin whu */
+	public NettyHttpInboundChannelAdapter(int port, String path) {
+		this(port, path, false);
+	}
+
+	public NettyHttpInboundChannelAdapter(int port, String path, boolean ssl) {
 		this.port = port;
 		this.ssl = ssl;
+		this.path = path;
+		putToChannelAdapter();
 	}
 
+	private void putToChannelAdapter() {
```

Comment:
Although deployments are currently single-threaded, it would be safer to synchronize this method.

Also, perhaps something like `ensurePathIsUnique()` would be a more suitable method name.